packages/mcp/src/configuration.test.ts

```diff
@@ -42,7 +48,10 @@ describe('MCPClient', () => {
   });
 
   beforeEach(async () => {
+    // Give each MCPClient a unique ID to prevent re-initialization errors across tests
+    const testId = expect.getState().currentTestName || `mcp-client-${Math.random().toString(36).substring(7)}`;
```

Comment:
you can give them all the same ID to create fewer subprocesses. Since you have the same config in each test you can do `afterEach->await mcp.disconnect()` so that it re-inits in each test